src/test/java/org/jabref/gui/mergeentries/FieldRowViewModelTest.java

```diff
@@ -0,0 +1,217 @@
+                .withField(StandardField.TITLE, "{Bridging the Heterogeneity of Orchestrations - A Petri Net-based Integration of BPEL and Windows Workflow}")
+                .withField(StandardField.ORGANIZATION, "IEEE")
+                .withField(StandardField.PAGES, "1--8")
+                .withField(StandardField.ADDRESS, "Oxford, United Kingdom")
+                .withField(StandardField.GROUPS, "By rating, Skimmed")
+                .withMonth(Month.DECEMBER)
+                .withField(StandardField.KEYWORDS, "a, b, c")
+                .withField(StandardField.YEAR, "2012");
+
+        mergedEntry = new BibEntry();
+        fieldMergerFactory = new FieldMergerFactory(mockedPrefs);
+
+        when(mockedPrefs.getGroupsPreferences()).thenReturn(mockedGroupsPrefs);
+        when(mockedGroupsPrefs.getKeywordSeparator()).thenReturn(',');
+    }
+
+    @Test
+    public void selectNonEmptyValueShouldSelectLeftFieldValueIfItIsNotEmpty() {
+        var numberFieldViewModel = createViewModelForField(StandardField.NUMBER);
+        numberFieldViewModel.selectNonEmptyValue();
+        assertEquals(FieldRowViewModel.Selection.LEFT, numberFieldViewModel.getSelection());
+
+        var authorFieldViewModel = createViewModelForField(StandardField.AUTHOR);
+        authorFieldViewModel.selectNonEmptyValue();
+        assertEquals(FieldRowViewModel.Selection.LEFT, authorFieldViewModel.getSelection());
+    }
+
+    @Test
+    public void selectNonEmptyValueShouldSelectRightFieldValueIfLeftValueIsEmpty() {
+        var monthFieldViewModel = createViewModelForField(StandardField.MONTH);
+        monthFieldViewModel.selectNonEmptyValue();
+        assertEquals(FieldRowViewModel.Selection.RIGHT, monthFieldViewModel.getSelection());
+
+        var addressFieldViewModel = createViewModelForField(StandardField.ADDRESS);
+        addressFieldViewModel.selectNonEmptyValue();
+        assertEquals(FieldRowViewModel.Selection.RIGHT, addressFieldViewModel.getSelection());
+    }
+
+    @Test
+    public void hasEqualLeftAndRightValuesShouldReturnFalseIfOneOfTheValuesIsEmpty() {
+        var monthFieldViewModel = createViewModelForField(StandardField.MONTH);
+        monthFieldViewModel.selectNonEmptyValue();
+        assertFalse(monthFieldViewModel.hasEqualLeftAndRightValues());
+    }
+
+    @Test
+    public void hasEqualLeftAndRightValuesShouldReturnTrueIfLeftAndRightAreEqual() {
+        var yearFieldViewModel = createViewModelForField(StandardField.YEAR);
+        yearFieldViewModel.selectNonEmptyValue();
+        assertTrue(yearFieldViewModel.hasEqualLeftAndRightValues());
+    }
+
+    /*
+// TODO: Fix this test
+    @Test
+    public void hasEqualLeftAndRightValuesShouldReturnTrueIfKeywordsAreEqual() {
+        var keywordsFieldViewModel = createViewModelForField(StandardField.KEYWORDS);
+        keywordsFieldViewModel.selectNonEmptyValue();
+        assertTrue(keywordsFieldViewModel.hasEqualLeftAndRightValues());
+    }*/
```

Comment:
Anything planned how to continue on that?

Reply:
I think in the future we should have multiple field row implementations. e.g. GroupsFieldRow, KeywordFieldRow, etc. And in each implementation, we override a method to compare left and right values. At first, I thought that the differences between field rows are minimal thus we can abstract it away in one class and pass factories as needed like in the FieldMerger case. But now, I can think of many specific cases where a field has special requirements. It is very unintuitive to keep passing factories.

For the time being, I'm going to keep the comment as a reminder to implement a different comparison logic based on the given field.

Reply:
Okay then add the test, annotate it with @Disabled and add a small comment on it